### PR TITLE
[CVE-2024-38526] Update Quarto to 1.4.556

### PR DESCRIPTION
**Pull request by ianpittwood**

### CODE REVIEW

**Comment by ianpittwood** [comment]
@aronatkins is there a preference on whether we should update the patch number on Quarto 1.4.x content images? Or should I keep what I have where I made a new content image with the patched Quarto version?

File: docker-bake.hcl [hunk 1]
```diff
@@ -158,6 +158,9 @@ variable CONTENT_BUILD_MATRIX {
 
       # R-4.4, Python-3.12, Quarto-1.4.
       {os = "ubuntu2204", os_alt = "jammy", r = "4.4.0", py = "3.12.3", drivers = "2024.03.0", quarto = "1.4.553"},
+
+      # R-4.4, Python-3.12.4, Quarto-1.4.556 (polyfill.js vulnerability patch)
+      {os = "ubuntu2204", os_alt = "jammy", r = "4.4.0", py = "3.12.4", drivers = "2024.03.0", quarto = "1.4.553"},
```
Comment (bschwedler) [inline docker-bake.hcl:163]: Should both of these be updated?
```suggestion
      # R-4.4, Python-3.12, Quarto-1.4.
      {os = "ubuntu2204", os_alt = "jammy", r = "4.4.0", py = "3.12.3", drivers = "2024.03.0", quarto = "1.4.556"},

      # R-4.4, Python-3.12.4, Quarto-1.4.556 (polyfill.js vulnerability patch)
      {os = "ubuntu2204", os_alt = "jammy", r = "4.4.0", py = "3.12.4", drivers = "2024.03.0", quarto = "1.4.556"},
```

**Comment by jspiewak** [comment]
Carlos updated Quarto again to ensure the embedded pandoc does not emit polyfill.io references. New version is 1.4.557

**Comment by aronatkins** [comment]
> @aronatkins is there a preference on whether we should update the patch number on Quarto 1.4.x content images? Or should I keep what I have where I made a new content image with the patched Quarto version?

Given how versions are treated today, it probably needs to be a distinct image.

**Review by bschwedler — APPROVED** [review]
LGTM

**Comment by dbkegley** [comment]
Reminder: we need to remember to also update the Connect helm chart default values on R/Python/Quarto installation changes

https://github.com/rstudio/helm/issues/531

https://github.com/rstudio/helm/blob/c66ac07fd5e7d18f4c9771c25d548c1e82ddf8fa/charts/rstudio-connect/values.yaml#L358

**Comment by ianpittwood** [comment]
> Reminder: we need to remember to also update the Connect helm chart default values on R/Python/Quarto installation changes
> 
> [rstudio/helm#531](https://github.com/rstudio/helm/issues/531)
> 
> https://github.com/rstudio/helm/blob/c66ac07fd5e7d18f4c9771c25d548c1e82ddf8fa/charts/rstudio-connect/values.yaml#L358

Can we create a `/opt/quarto/default` symlink instead? Or perhaps we could eliminate putting a version in the path altogether in this case? It doesn't seem like multiple versions of Quarto should need to be installed.

**Pull request merged**